examples/plot_synthetic.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+"""
+==========================
+Synthetic dataset example with visualization
+==========================
+"""
+
+# %%
+from fairlearn.datasets import make_synthetic_dataset
+from fairlearn.metrics import (
+    MetricFrame,
+    false_positive_rate,
+    true_positive_rate,
+    selection_rate,
+    count
+)
+import numpy as np
+from sklearn.metrics import accuracy_score, precision_score, recall_score
+from sklearn.model_selection import train_test_split
+from sklearn.tree import DecisionTreeClassifier
+
+
+rng = np.random.RandomState(seed=42)
+X, y, gender = make_synthetic_dataset(rng=rng)
+
+X_train, X_test, y_train, y_test, gender_train, gender_test = train_test_split(
+    X, y, gender, test_size=0.3, random_state=rng
+)
+
+classifier = DecisionTreeClassifier(min_samples_leaf=10, max_depth=4)
+classifier.fit(X, y)
```

Comment:
We should probably use `X_train, y_train` here and predict using `X_test`?
And then evaluate the metrics on `y_test` and `gender_test`.

Reply:
Indeed! I didn't think carefully about this at all when I copied things from the existing notebook. It's fixed now.